src/poetry/publishing/uploader.py

```diff
@@ -225,6 +224,10 @@ def _upload_file(
     ) -> None:
         from cleo.ui.progress_bar import ProgressBar
 
+        if not file.is_file():
+            self._io.write_error(f"Archive ({file}) does not exist", True)
```

Comment:
I don't know that this `write_error` is needed -- the error message should be communicated to the user just through `raise`, right?

Reply:
It should error out to the user but in the previous comment, it was mentioned to write it via `self._io`.

Reply:
The comment above meant the two options to be mutually exclusive -- that is to say, are we better off dropping everything on the floor (`raise`), or writing a message (`self._io`) and explaining we ran into error at the end (non-zero status code). I'm inclined to lean towards the first without any strong feelings from anyone else.